app/logic/solana_logic.rb

```diff
@@ -495,8 +497,12 @@ def cli_request(cli_method, rpc_urls)
       end
 
       # byebug
-
-      return JSON.parse(response_utf8) unless response_utf8 == ''
+      if response_utf8 != ''
+        # Include errors from the solana-cli 
+        return JSON.parse(response_utf8).merge!('cli_error' => error_msg)
+      elsif !error_msg.blank?
+        return {cli_error: error_msg}
```

Comment:
In one scenario you return a JSON string, in the other you return a Hash. Please choose one and make them consistent.